crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -261,6 +262,7 @@ impl SlidingSyncBuilder {
             rooms,
 
             position: StdRwLock::new(SlidingSyncPositionMarkers { pos: None, delta_token }),
+            past_positions: StdRwLock::new(RingBuffer::new(20)),
```

Comment:
Not that it matters much, but I would even go with a smaller size, like 4 or 8, since we shouldn't have to remember positions that happened a long time ago (unless we strongly doubt that we could receive very old pos responses).

Reply:
`SlidingSyncPositionMarkers` isn't very weight, I think 20 is good if the network has some high latency. We never know :-p.